contracts/contracts/sbtc-token.clar

```diff
@@ -110,6 +112,22 @@
 	)
 )
 
+(define-public (transfer-many (recipients (list 200 { amount: uint, sender: principal, to: principal, memo: (optional (buff 34)) })))
+	(fold complete-individual-transfer recipients (ok u0))
+)
+
+(define-private (complete-individual-transfer (individual-transfer { amount: uint, sender: principal, to: principal, memo: (optional (buff 34)) }) (helper-response (response uint uint)))
+    (match helper-response 
+        index
+            (begin 
+                (unwrap! (transfer (get amount individual-transfer) (get sender individual-transfer) (get to individual-transfer) (get memo individual-transfer)) (err (+ ERR_TRANSFER_INDEX_PREFIX (+ u10 index))))
```

Comment:
This is largely just food for thought, but figured I'd throw out the idea anyway --

It looks like we don't preserve the eventual error code from `transfer()` with this, right? I wonder if we could encode both the transfer index and the transfer error code by assigning a range for transfers:

```
(define-constant ERR_TRANSFER_INDEX_PREFIX u1000)

(define-private (complete-individual-transfer (individual-transfer { amount: uint, sender: principal, to: principal, memo: (optional (buff 34)) }) (helper-response (response uint uint)))
    (match helper-response 
        index
            (let ((transfer-result (transfer (get amount individual-transfer) (get sender individual-transfer) (get to individual-transfer) (get memo individual-transfer))))
                (match transfer-result
                    transfer-ok (ok (bit-or (bit-shift-left (+ ERR_TRANSFER_INDEX_PREFIX index) u4) u0))
                    transfer-err (err (bitwise-or (bit-shift-left (+ ERR_TRANSFER_INDEX_PREFIX index) u4) transfer-err))
                )
            )
        err-response
            (err err-response)
    )
)
´´´

Which I think should give something like:

```
Index: 0
Result: ok
Encoded Value: 10000 (1000 << 4 | 0)

Index: 1
Result: ERR_NOT_OWNER
Encoded Value: 10011 (1001 << 4 | 1)

Index: 2
Result: ok
Encoded Value: 10020 (1002 << 4 | 0)
```

Reply:
Right. I see the idea. we use this fold method a few other times in this contract so if we change it here we should change it everywhere else. 

I'll open an issue for this as a 'nice-to-have' & think on how I'd implement.

Reply:
What is the purpose of returning the index in the error?  Is it so that other contracts can fix the call? 

If the index is only about user experience, tools like stxer.xyz can help users to understand errors.  

I'd prefer a simpler call with just a match and (transfer..)

Reply:
Yeap. So that an operator can quickly remove the errored transaction & re-submit.

Reply:
Yeah this was primarily just an idea on how to give the caller clear information directly in the result (based on what they submitted), and thinking that the errors for transfer may expand in the future. But just an idea, I'll resolve this 😊

Reply:
For me, the extra costs are not justified for the convenience of the user